web/hooks/pull_request_test.go

```diff
@@ -0,0 +1,29 @@
+package hooks
+
+import "testing"
+
+func TestGetLinkedIssue(t *testing.T) {
+	var wantIssueNumber uint64 = 30
+	tests := map[string]struct {
+		body        string
+		expectError bool
+	}{
+		"Simple":         {body: "Fixes #30", expectError: false},
+		"Not a number":   {body: "Fixes #30nan", expectError: true},
+		"Invalid body":   {body: "Fixes #30nan #", expectError: true},
+		"Invalid body 2": {body: "Fixes", expectError: true},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			gotIssueNumber, err := getLinkedIssue(tt.body)
+			if err != nil {
```

Comment:
This captures the case when we get an error and it is expected. We should also capture the case when we get an error and it is unexpected, e.g., something like:
```go
if err != nil {
  if tt.expectError {
    return
  }
  t.Errorf("getLinkedIssue() = %v, no error expected", err)
}
```
On the other hand, it might be easier to just make the map struct provide the expected error as a string:
```go
tests := map[string]struct {
	body   string
	errStr string
}{
	"Simple": {body: "Fixes #30", errStr: ""},
	"Not a number": {body: "Fixes #30nan", err: "failed to parse issue number..."},
}
```
You could also use `error` instead of string, but then it is tricker to create the correct error...